homeassistant/util/loop.py

```diff
@@ -42,28 +48,48 @@ def raise_for_blocking_call(
     offender_filename = offender_frame.f_code.co_filename
     offender_lineno = offender_frame.f_lineno
     offender_line = _get_line_from_cache(offender_filename, offender_lineno)
+    report_key: tuple[str | None, str, int | Any]
 
     try:
         integration_frame = get_integration_frame()
     except MissingIntegrationFrame:
         # Did not source from integration? Hard error.
+        report_key = (None, offender_filename, offender_lineno)
+        was_reported = report_key in _PREVIOUSLY_REPORTED
+        _PREVIOUSLY_REPORTED.add(report_key)
         if not strict_core:
-            _LOGGER.warning(
-                "Detected blocking call to %s with args %s in %s, "
-                "line %s: %s inside the event loop; "
-                "This is causing stability issues. "
-                "Please create a bug report at "
-                "https://github.com/home-assistant/core/issues?q=is%%3Aopen+is%%3Aissue\n"
-                "%s\n"
-                "Traceback (most recent call last):\n%s",
-                func.__name__,
-                mapped_args.get("args"),
-                offender_filename,
-                offender_lineno,
-                offender_line,
-                _dev_help_message(func.__name__),
-                "".join(traceback.format_stack(f=offender_frame)),
-            )
+            if was_reported:
+                _LOGGER.debug(
+                    "Detected blocking call to %s with args %s in %s, "
+                    "line %s: %s inside the event loop; "
+                    "This is causing stability issues. "
+                    "Please create a bug report at "
+                    "https://github.com/home-assistant/core/issues?q=is%%3Aopen+is%%3Aissue\n"
+                    "%s\n",
+                    func.__name__,
+                    mapped_args.get("args"),
+                    offender_filename,
+                    offender_lineno,
+                    offender_line,
+                    _dev_help_message(func.__name__),
+                )
+            else:
+                _LOGGER.warning(
```

Comment:
Sadly we have to branch here because we don't want the traceback with debug.  I thought about appending the Traceback to the message every time but it seemed harder to follow the code so I left it as two messages.